contracts/registry/PocketRegistryDelegate.sol

```diff
@@ -28,131 +29,95 @@ contract PocketRegistryDelegate {
 
     tokenAddress.call(bytes4(sha3("burn(uint256,address)")),1,msg.sender);
 
-    PocketNode newNode = new PocketNode();
-    // TODO: nodeDelegateAddress should only get set once per upgrade
-    newNode.supportedTokens(_supportedTokens);
-    newNode.url(_url);
-    newNode.port(_port);
-    newNode.isRelayer(_isRelayer);
-    newNode.isOracle(_isOracle);
-    newNode.changeDelegate(nodeDelegateAddress);
-    newNode.setOwner(msg.sender);
-    newNode.setTokenAddress(tokenAddress);
+    PocketNode newNode = new PocketNode(msg.sender, nodeDelegateAddress, tokenAddress, _isRelayer, _isOracle);
     userNode[msg.sender] = newNode;
 
-    registeredNodes.push(newNode);
-    registerNode(newNode.address, newNode.supportedTokens, newNode.url, newNode.port, newNode.isRelayer, newNode.isOracle);
+    registerNode(newNode.address, _supportedTokens, _url, _port, _isRelayer, _isOracle);
```

Comment:
This should be: `registerNode(newNode, _supportedTokens, _url, _port, _isRelayer, _isOracle);`

Reply:
Can't pass the newNode as a parameter since is a PocketNode model